File: TypeScript/src/drawing/shapes/triangle.ts
import { Line } from "./line";
import { AbstractShape } from "./shape";

/**
 * Refactor Exercise 3: (Mis-)Shaped
 *
 * @author Zishen Wen (F22), Deyuan Chen (S22)
 */
class Triangle extends AbstractShape {

    private points: number[][];

    constructor(point1: number[], point2: number[], point3: number[]) {
      super();
        this.points = [
			point1, point2, point3
		]
    }

    public toLines(): Line[] {
        return [
			new Line(this.points[0], this.points[1]),
			new Line(this.points[1], this.points[2]),
			new Line(this.points[2], this.points[0])
		]
    }
}

export { Triangle }

File: Java/src/main/java/drawing/shapes/Triangle.java
package drawing.shapes;

/**
 * Refactor Task 3: (Mis-)Shaped
 *
 * @author Zishen Wen (F22), Deyuan Chen (S22)
 */
public class Triangle implements Shape {

    private final int[][] points;

    public Triangle(int[] point1, int[] point2, int[] point3) {
        this.points = new int[][]{
                point1, point2, point3
        };
    }

    @Override
    public Line[] toLines() {
        return new Line[]{
                new Line(this.points[0], this.points[1]),
                new Line(this.points[1], this.points[2]),
                new Line(this.points[2], this.points[0])
        };
    }
}
